### PR TITLE
Support new encodings for enums

### PR DESCRIPTION
**Pull request by ThomasDebrunner**
This adds support for

- New enum encoding: https://github.com/Auterion/libmav/issues/41
- XML files with only enum declarations (no messages)

### CODE REVIEW

File: include/mav/MessageSet.h [hunk 1]
```diff
@@ -70,6 +83,53 @@ namespace mav {
                 _document(std::move(document)),
                 _root_xml_folder_path(root_xml_folder_path) {}
 
+        
+        static inline uint64_t _strict_stoul(const std::string &str, int base=10) {
+            if (str.empty()) {
+                throw ParseError("Enum value is empty string");
+            }
+
```
Comment (KonradRudin) [inline include/mav/MessageSet.h:91]: At the moment, you could input a negative value, which gets convered to an unsigned. You could check here as well that the string only contains numbers/hex values?

File: include/mav/MessageSet.h [hunk 2]
```diff
@@ -70,6 +83,53 @@ namespace mav {
...
+                return res;
+            } catch (std::exception &e) {
+                throw ParseError("Could not parse " + str + " as a number (stoul failed): " + e.what());
+            }
+        }
+
+        static uint64_t _parseEnumValue(const std::string &str) {
+            // Check for binary format: 0b or 0B
+            if (str.size() >= 2 && (str.substr(0, 2) == "0b" || str.substr(0, 2) == "0B")) {
+                return std::bitset<64>(str.substr(2)).to_ullong();
```
Comment (KonradRudin) [inline include/mav/MessageSet.h:107]: Question for me, why aren't you using here the _strict_stoul with base 2?

File: include/mav/MessageSet.h [hunk 3]
```diff
@@ -70,6 +83,53 @@ namespace mav {
...
+            }
+
+            // Check for hexadecimal format: 0x or 0X
+            if (str.size() >= 2 && (str.substr(0, 2) == "0x" || str.substr(0, 2) == "0X")) {
+                return _strict_stoul(str, 16);
```
Comment (KonradRudin) [inline include/mav/MessageSet.h:112]: you could get rid of the first 2 characters?

File: include/mav/MessageSet.h [hunk 4]
```diff
@@ -127,7 +187,11 @@ namespace mav {
             auto istream = std::istringstream(xml_string);
             auto file = std::make_shared<rapidxml::file<>>(istream);
             auto doc = std::make_shared<rapidxml::xml_document<>>();
-            doc->parse<0>(file->data());
+            try {
+                doc->parse<0>(file->data());
+            } catch (rapidxml::parse_error &e) {
```
Comment (KonradRudin) [inline include/mav/MessageSet.h:192]:
```suggestion
            } catch (const rapidxml::parse_error &e) {
```

File: tests/MessageSet.cpp [hunk 5]
```diff
@@ -109,6 +144,89 @@ TEST_CASE("Create messages from set") {
...
+    <enums>
+        <enum name="MY_ENUM_HAPPY_PATH">
+            <entry value="1" name="BIT0" />
+            <entry value="2**4" name="BIT4" />
+            <entry value="0b000100000000" name="BIT8" />
+            <entry value="0x10000" name="BIT16" />
+            <entry value="0b1000000000000000000000000000000000000000000000000000000000000" name="BIT60" />
+            <entry value="2305843009213693952" name="BIT61" />
+            <entry value="2**62" name="BIT62" />
+            <entry value="0x8000000000000000" name="BIT63" />
```
Comment (KonradRudin) [inline tests/MessageSet.cpp:164]:
```suggestion
            <entry value="0B1000000000000000000000000000000000000000000000000000000000000" name="BIT60" />
            <entry value="2305843009213693952" name="BIT61" />
            <entry value="2**62" name="BIT62" />
            <entry value="0X8000000000000000" name="BIT63" />
```

Comment (KonradRudin) [reply tests/MessageSet.cpp:164]: Just to have both upper and lower case checked.

File: include/mav/MessageSet.h (hunk 2 above)
Comment (ThomasDebrunner) [reply include/mav/MessageSet.h:107]: Done

File: include/mav/MessageSet.h (hunk 3 above)
Comment (ThomasDebrunner) [reply include/mav/MessageSet.h:112]: Done

**Pull request merged**